compiler-core/src/exhaustiveness.rs

```diff
@@ -441,7 +442,7 @@ pub enum RuntimeCheck {
         args: Vec<Variable>,
     },
     EmptyList,
-    NonEmptyList {
+    List {
```

Comment:
Undo this change, it's under-specified now. Both the empty and non-empty variants are lists, and this name fails to specify what is distinctive about this variant.

A list pattern would check that it is any list, not that it is a non-empty list.